### PR TITLE
VAM-787: Create multi-vista synthetic data

### PR DESCRIPTION
**Pull request by creindle**
- Condensed scripts allows for excess entries to be created for all clinical collections
- Ensured created entries had Patient ids that were queried based on the input localStationNumber
- Ensured created Allergies had GmrAllergies that were queried based on the input localStationNumber

### CODE REVIEW

File: scripts/generateData.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    const inputCollectionName = argv.c || 'Vital';
+    const inputQuantity = argv.q || 1;
+    generateSynthData(inputCollectionName, inputStationId, inputQuantity, (err, res) => {
+        if (err) console.log(err);
+        return res;
+    });
+};
+
+if (generateVista === 'multi') {
+    generateVistaData('paloAlto');
```
Comment (creindle) [inline scripts/generateData.js]: I just realized, when I'm calling these two functions, they need to be called so they run sequentially. I'm fixing this now.

**Comment by mfuroyama** [comment]
The code looks okay, but regarding the run-time parameters: you might want to consider implementing some kind of configuration scheme instead of hard-coding the parameters into the source code.

Consider implementing some sort of JSON-style configuration file scheme to pass the parameters into the script. Basically, do what you're doing, but instead of hard-coding the values, move the values out into a JSON config file, and have your code read the file for the values.

Let me know if there's any confusion or if you have any questions about this comment

File: scripts/populateDBData.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+const _ = require('lodash');
+const async = require('async');
+const mongo = require('mongodb');
+const templates = require('./collectionTemplates');
+const utils = require('../utils/utils');
+
+const MongoClient = mongo.MongoClient;
+const mongoOptions = {
```
Comment (creindle) [inline scripts/populateDBData.js]: I need to replace these variables with the info in the JSON files.

File: syntheticDataSchemas/cheyenne.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+	"MONGOOPTIONS": {
+	    "hostname": "localhost",
+	    "port": 27017
+	},
+	"MONGODATABASE": "vam",
```
Comment (mfuroyama) [inline syntheticDataSchemas/cheyenne.json]: This is fine, but just out of pure curiosity, what made you put the `MONGODATABASE` field outside of the `MONGOOPTIONS` object?

File: scripts/generateData.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+const generateCustomData = () => {
+    if (!argv.u || !argv.s || !argv.c || !argv.q) return console.log(error.missingCustomFlag);
+    const inputStationId = argv.s;
+    const inputCollectionName = argv.c;
+    const inputQuantity = argv.q;
+    if (!_.isInteger(inputStationId)) return console.log(error.incorrectStationId);
+    else if (!_.isInteger(inputQuantity)) return console.log(error.incorrectQuantity);
+    else if (!_.includes(['Vital', 'Allergy', 'NewPerson', 'Patient', 'Problem'], inputCollectionName)) return console.log(error.incorrectCollectionName);
+    generateSynthData(mongoOptions, mongoDatabase, inputCollectionName, inputStationId, inputQuantity, (err, res) => {
+        if (err) return console.log(err);
```
Comment (creindle) [inline scripts/generateData.js]: I noticed the terminal hangs if there is an error here. I'm a little bit stumped as to what the solution may be. I'm thinking this needs to be changed to an async function so the code understands when to end.

File: syntheticDataSchemas/cheyenne.json (hunk 3 above)
Comment (creindle) [reply syntheticDataSchemas/cheyenne.json]: The original implementation had the mongoDatabase outside of the mongo options, so I was keeping the format the same.

File: scripts/generateData.js (hunk 4 above)
Comment (creindle) [reply scripts/generateData.js]: For future references (for myself) this was due to the async waterfall breaking out of the chain and, therefore, not reaching the last function to close the mongo connection.

**Comment by mfuroyama** [comment]
One last favor to ask. Could you rebase your code to the latest `master` branch and run the test one more time? Remember how we were talking about how some of the "national" data types actually turned into "local" data types. I think this will probably affect the writes of the clinical data. Either way, could you double-check and repair accordingly?  Thanks, @creindle!!

**Comment by mfuroyama** [comment]
Looks good! Two *really* small, slightly pedantic things to consider:
1. Really minor, but let's check the formatting and indentation of the configuration JSON files. It looks slightly off, at least in the GitHub code browser view.
2. Again, really minor, but consider wrapping the final line of `generateData.js` with a conditional that checks if you've invoked the module from the command line. So:
```Javascript
if (require.main === module) {
    generateVistaData(generateVista);
}
```

This ensure that the `generateVistaData` function is only called if you invoke the code "executable" style.

**Pull request merged**